src/interpreter.c

```diff
@@ -110,6 +137,41 @@ static int play_bar_token(Player* player, Instrument* instrument, BarToken* bar)
 	return Player_play_bar(player);
 }
 
+static int print_bar_length_error(const Interpreter* interpreter, FILE* stream, const BarToken* bar, int error_code) {
+
+	int status = 0;
+
+	if(interpreter == NULL || stream == NULL || bar == NULL) {
+		return ERROR_CODE_INVALID_ARGUMENT;
+	}
+
+	if(bar->content == NULL) {
+		return ERROR_CODE_INVALID_STATE;
+	}
+
+	fprintf(stream, "%s:%u:%u: ", interpreter->filename, bar->line, bar->col);
+
+	switch(error_code) {
+
+		case ERROR_CODE_BAR_TOO_SHORT:
+			fputs("Bar too short:\n", stream);
+			break;
+
+		case ERROR_CODE_BAR_TOO_LONG:
+			fputs("Bar too long:\n", stream);
+			break;
+
```

Comment:
Probably better to have some kind of `strerror`-like mapping between error codes and messages inside the `error` module.

Reply:
(Implemented)